com.unity.ml-agents/Runtime/Inference/GeneratorImpl.cs

```diff
@@ -346,13 +346,13 @@ public void Generate(TensorProxy tensorProxy, int batchSize, IEnumerable<AgentIn
     /// It will use the Texture input data contained in the agentInfo to fill the data
     /// of the tensor.
     /// </summary>
-    internal class VisualObservationInputGenerator : TensorGenerator.IGenerator
+    internal class NonVectorObservationInputGenerator : TensorGenerator.IGenerator
```

Comment:
`NDimObservationInputGenerator`?

Reply:
NDim could be N = 1. I thought it would make more sense to do Vector and non-Vector, but I can change it

Reply:
MultiDim?
Non-Vector just sounds very broad and not very self explanatory but I'm not hard opposed to this

Reply:
We could probably actually unify Vector+VisualObservationInputGenerator - I think they do almost the same thing, except that Vector can use multiple sensors to write to the output, and Visual only ever has one